contracts/strategies/StrategyUSDT3pool.sol

```diff
@@ -0,0 +1,431 @@
+    uint public threshold = 6000;
+    uint public slip = 100;
+    uint public tank = 0;
+    uint public p = 0;
+
+    event Threshold(address indexed strategy);
+
+    modifier isAuthorized() {
+        require(msg.sender == governance || 
+                msg.sender == strategist || 
+                msg.sender == controller ||
+                msg.sender == address(this), "!authorized");
+        _;
+    }
+
+    constructor(address _controller) public {
+        governance = msg.sender;
+        strategist = msg.sender;
+        controller = _controller;
+    }
+    
+    function getName() external pure returns (string memory) {
+        return "StrategyUSDT3pool";
+    }
+    
+    function deposit() public isAuthorized {
+        rebalance();
+        uint _want = (IERC20(want).balanceOf(address(this))).sub(tank);
+        if (_want > 0) {
+            IERC20(want).safeApprove(_3pool, 0);
+            IERC20(want).safeApprove(_3pool, _want);
+            uint v = _want.mul(1e30).div(ICurveFi(_3pool).get_virtual_price());
+            ICurveFi(_3pool).add_liquidity([0, 0, _want], v.mul(DENOMINATOR.sub(slip)).div(DENOMINATOR));
+        }
+        uint _bal = IERC20(_3crv).balanceOf(address(this));
+        if (_bal > 0) {
+            IERC20(_3crv).safeApprove(y3crv, 0);
+            IERC20(_3crv).safeApprove(y3crv, _bal);
+            yvERC20(y3crv).deposit(_bal);
+        }
+    }
+    
+    // Controller only function for creating additional rewards from dust
+    function withdraw(IERC20 _asset) external returns (uint balance) {
+        require(msg.sender == controller, "!controller");
+        require(want != address(_asset), "want");
+        require(_3crv != address(_asset), "3crv");
+        require(y3crv != address(_asset), "y3crv");
+        balance = _asset.balanceOf(address(this));
+        _asset.safeTransfer(controller, balance);
+    }
+    
+    // Withdraw partial funds, normally used with a vault withdrawal
+    function withdraw(uint _amount) external {
+        require(msg.sender == controller, "!controller");
+
+        rebalance();
+        uint _balance = IERC20(want).balanceOf(address(this));
+        if (_balance < _amount) {
+            _amount = _withdrawSome(_amount.sub(_balance));
+            _amount = _amount.add(_balance);
+            tank = 0;
+        }
+        else {
+            if (tank >= _amount) tank = tank.sub(_amount);
+            else tank = 0;
+        }
+
+        address _vault = Controller(controller).vaults(address(want));
+        require(_vault != address(0), "!vault"); // additional protection so we don't burn the funds
+        uint _fee = _amount.mul(withdrawalFee).div(DENOMINATOR);
+        IERC20(want).safeTransfer(Controller(controller).rewards(), _fee);
+        IERC20(want).safeTransfer(_vault, _amount.sub(_fee));
+    }
+
+    function _withdrawSome(uint _amount) internal returns (uint) {
+        uint _amnt = _amount.mul(1e30).div(ICurveFi(_3pool).get_virtual_price());
```

Comment:
Comparing this against the StrategyTUSDypool file, and you use here exponent 30, where in the TUSD you use 18, this is because of the USDT 6 decimals i assume right?

```
function _withdrawSome(uint256 _amount) internal returns (uint256) {
        uint256 _amnt = _amount.mul(1e18).div(ICurveFi(ypool).get_virtual_price());
        uint256 _amt = _amnt.mul(1e18).div(yvERC20(yycrv).getPricePerFullShare());
       ...
    }
```

Reply:
Exactly. 18 and 6 have 12 difference. so 18 turns into 30.